operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -1609,6 +1690,91 @@ func (rc *ReconciliationContext) CheckRollingRestart() result.ReconcileResult {
 	return result.Continue()
 }
 
+func (rc *ReconciliationContext) MaybeUpdateCondition(condition api.DatacenterCondition) bool {
```

Comment:
how about just `setCondition()`?

Reply:
Because it is conditional? We already have a `SetCondition()` method on the Datacenter resource.

Reply:
but I think it's fine that it's conditional - just add a docstring and keep the return value the same, no need for the `maybe`

Reply:
What about the similar names between the `SetCondition` on the Datacenter and the proposed `setCondition` method here? Is that confusing?